RMextract/getRM.py

```diff
@@ -1,144 +1,161 @@
 import os
-import numpy as np
 from datetime import date
+from pathlib import Path
+from typing import Callable, List, Literal, Optional, Tuple, Union
+
+import numpy as np
 
+from RMextract import PosTools
+from RMextract import getIONEX as ionex
 from RMextract.EMM import EMM as EMM
+from RMextract.formatters import Formatter
+from RMextract.logging import logger
 
 ION_HEIGHT=PosTools.ION_HEIGHT
 #####################  main processing function #####################
-def getRM(MS=None,
-           server="ftp://ftp.aiub.unibe.ch/CODE/",
-           prefix='codg',
-           ionexPath="IONEXdata/",
-           earth_rot=0,
-           timerange=0,
-           use_azel = False,ha_limit=-1000,use_filter=None,
-           **kwargs):
-    '''optional arguments are:
-    radec or pointing : [ra,dec] in radians, or if use_azel =True, az + el in radians,
-    timestep in s, timerange = [start, end]in MJD seconds, 
-    otherwise use start_time/end_time (casa timestring,eg. 2012/11/21/12:00:00  or 56252.5d  NEEDS PYRAP) ,
-    stat_names = list of strings per station, 
-    stat_positions = list of length 3 numpy arrays, containing station_position in ITRF meters.
-    useEMM = boolean, use EMM for Earth magnetic field, otherwise WMM cooefficients will be used.
-    out_file = string, if given the data points will be written to a text file.
-    use_mean = True if you only want report for mean of station positions
-    use_filter =standard deviation,or list of standard deviations (time,long, lat) to gaussian filter TEC data 
-    TIME_OFFSET = float, offset time at start and end to ensure all needed values are calculated,
-    overwrite = boolean, if True overwrite existing IONEX files and download them again,
-    Returns the (timegrid,timestep,TEC) where TEC is a dictionary containing 1 enumpyarray per station in stat_names. 
-    If stat_names is not given, the station names will either be extracted from the MS or st1...stN '''
 
-    #print ('earth_rot', earth_rot)
-    #print ('timerange ', timerange)
-    #print ('use_azel ', use_azel)
-    stat_names=[]
-    useEMM=False
-    use_mean = False
-    myobject = ''
-    timestep=60
-    pointing=[0.,0.5*np.pi]
-    out_file=''
-    stat_pos=[PosTools.posCS002]
-    use_proxy = False
-    overwrite = False
-    #stat_names=['LOFAR_CS002']
-    if not (MS is None):
+def getRM(
+    MS: Optional[str]=None,
+    server="ftp://ftp.aiub.unibe.ch/CODE/",
+    prefix='codg',
+    ionexPath="IONEXdata/",
+    earth_rot: float=0,
+    timerange: Union[List[float], Literal[0]]=0,
+    start_time: Optional[str] = None,
+    end_time: Optional[str] = None,
+    pointing: List[float] = [0.,0.5*np.pi],
+    radec: Optional[List[float]] = None,
+    use_azel = False,
+    ha_limit=-1000,
+    use_filter: Optional[Union[float, List[float]]] = None,
+    use_urlib = False,
+    formatter: Optional[Formatter]=None,
+    use_mean: Optional[bool] = None,
+    stat_names=[],
+    useEMM=False,
+    object = '',
```

Comment:
Agreed - this argument was copied from the existing hidden kwarg:
https://github.com/lofar-astron/RMextract/blob/25638b1f8544083c388e0c699cfeb503ba7b8147/RMextract/getRM.py#L59

Happy to change, but it could break existing use elsewhere.a